### PR TITLE
feat: automatic code style checking

### LINKED ISSUE #108: Require conformance to style standards
#### Issue Summary

In the `CONTRIBUTING.md` file there is a section on Style Guidelines & Naming Conventions. Please review and implement a check in Travis CI. If there are any style errors, either fix or create a rule to exclude the error.

Before writing any code, we suggest you review our `CONTRIBUTING.md` file and propose your solution for discussion here in the comments. Thank you!

#### Aceptance Criteria

* TravisCI checks that the code conforms to our Style Guidelines & Naming Conventions as defined in our `CONTRIBUTING.md` file
* tests pass locally, in Travis CI and using Docker

### PR DESCRIPTION
**Pull request by misantron**
# Fixes # 

Closes #108 

### Checklist
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation about the functionality in the appropriate .md file
- [ ] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Run code style authomatic checking (PSR-1, PSR-2) on TravisCI build
- Make it possible to run unit tests and CS check with Composer commands
- Some fixes in TravisCI project configuration

### CODE REVIEW

**Comment by thinkingserious** [comment]
Hello @misantron,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

File: lib/Client.php [hunk 1]
```diff
@@ -349,9 +356,9 @@ private function createCurlOptions($method, $body = null, $headers = null)
         $options = [
                 CURLOPT_RETURNTRANSFER => true,
                 CURLOPT_HEADER => true,
-                CURLOPT_CUSTOMREQUEST => strtoupper($method),
+                CURLOPT_CUSTOMREQUEST => mb_strtoupper($method),
```
Comment (peter279k) [inline lib/Client.php]: If we want to use the `mb_strtoupper`, we have to edit `composer.json` and add the `ext-mbstring: "*"` in `require` block.

File: phpunit.xml.dist [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    </testsuites>
+    <filter>
+        <whitelist processUncoveredFilesFromWhitelist="true">
+            <directory suffix=".php">lib</directory>
+            <exclude>
+                <directory suffix="Interface.php">src</directory>
+            </exclude>
+        </whitelist>
+    </filter>
+</phpunit>
```
Comment (peter279k) [inline phpunit.xml.dist]: It seems that this file misses EOF(end of file). Please add this.

File: .gitignore [hunk 3]
```diff
@@ -8,4 +8,9 @@ vendor/
...
 
 # Environment files
-.env/*.*
\ No newline at end of file
+.env/*.*
+
+# Local configs
+.php_cs
+.php_cs.cache
+phpunit.xml
```
Comment (peter279k) [inline .gitignore]: It seems that this file misses EOF(end of file). Please add this.

**Comment by misantron** [comment]
@peter279k made fixes for ext-mbstring dependency and EOF in gitignore and phpunit.xml.dist

**Comment by thinkingserious** [comment]
Hello @misantron,<br /><br/>Thanks again for the PR!<br /><br />                                         We noticed that you have earned 3 out of the 5 points needed to receive glorious SendGrid Hacktoberfest swag.<br /><br />                                         Please take a moment to checkout <a href="https://dx.sendgrid.com/hacktoberfest/issues">this link</a> to find more issues to get you past the required threshold.<br /><br />                                         Also, please be sure that you have officially registered <a href="https://dx.sendgrid.com/hacktoberfest/join">here</a>.<br /><br />                                         Thank you and Happy Hacktobering!<br /><br />                                         Team SendGrid DX

**Comment by thinkingserious** [comment]
Hello @misantron,<br /><br/>Thanks again for the PR!<br /><br />                                         You have earned 3 out of the 5 points needed to receive glorious SendGrid Hacktoberfest swag.<br /><br />                                         Please take a moment to checkout <a href="https://dx.sendgrid.com/hacktoberfest/issues">this link</a> to find more issues to get you past the required threshold or to simply continue the celebration.<br /><br />                                         Also, <strong>please be sure you have officially registered with us <a href="https://dx.sendgrid.com/hacktoberfest/join">here</a> by November 1, 2018</strong> to qualify.<br /><br />                                         If you have any questions you can email us at <a href="mailto:dx+hacktoberfest2018@sendgrid.com">dx+hacktoberfest2018@sendgrid.com</a>.<br /><br />                                         Thank you and Happy Hacktobering!<br /><br />                                         Team SendGrid

**Pull request merged**